virtio-devices/src/net.rs

```diff
@@ -565,6 +566,12 @@ impl Net {
             taps.push(tap);
         }
 
+        if let Some(mtu) = mtu {
+            if !taps.is_empty() {
+                taps[0].set_mtu(mtu as i32).map_err(Error::TapError)?;
```

Comment:
Only setting on the first one? What about the others?

Reply:
Also, you have a code that checks if `taps` is empty and if so conditionally supply the feature. Can taps ever actually be empty? Should it not instead error out.

Reply:
Isn't the MTU at the interface level? I mean we don't set IP and MAC address on each queue, only on the first one.

Reply:
> Also, you have a code that checks if `taps` is empty and if so conditionally supply the feature. Can taps ever actually be empty? Should it not instead error out.

Sure we could simplify the whole thing by returning an error if it's actually empty, and after that we can assume it's not empty.

Reply:
> Sure we could simplify the whole thing by returning an error if it's actually empty, and after that we can assume it's not empty.

This is what I would do - the vector shouldn't be empty.